nifi-stateless/nifi-stateless-api/src/main/java/org/apache/nifi/stateless/engine/StatelessEngineConfiguration.java

```diff
@@ -89,4 +90,20 @@ default boolean isLogExtensionDiscovery() {
      * A <code>null</code> value indicates that no status tasks are scheduled.
      */
     String getStatusTaskInterval();
+
+    /**
+     * @return a String representing the number of milliseconds that the process scheduler should wait for a process to start
+     * Defaults to 10 seconds or 10_000 milliseconds
+     */
+    default long getProcessorStartTimeout() {
```

Comment:
Should take an argument for the `TimeUnit` to use

Reply:
Using `java.time.Duration` could provide another more flexible approach that avoids confusion related to time units.